### DISCUSSION TITLE
Importing routing.yaml from bundle (new dir structure)

### DISCUSSION THREAD

**Discussion by Niiph** [symfony/symfony#49468] [Q&A]
According to this link: 
[Symfoy 6.2 docs](https://symfony.com/doc/current/bundles.html#bundle-directory-structure)
all bundle config files should be placed inside `BundleName/config` directory, but after doing this it is not possible to import `routing.yaml` into `config/routes.yaml` in projekt which uses this bundle.  
Is there some trick to it?

**Comment by ahundiak** [comment]
Take a look at the config/routes/web_profiler.yaml file for an example.  Something like:
```
my_bundle:
    resource: '@MyBundle/config/routes.yaml'
```
Don't overlook the `@`.  If you are trying to do something else then post what you tried.

**Reply by Niiph** [reply]
@ahundiak This is exactly what I was trying to do:
```yaml
_async_task_bundle:
    resource: '@xxxAsyncTaskBundle/config/routing.yaml'
```
And this will not work because you can't import from config directory. 
Example you suggest is based on old bundle configuration/directory structure, where everything was inside main directory

**Reply by ahundiak** [reply]
I tested before posting and it works just fine.  Nothing really special about config.  Are you using `bin/console debug:router` to check for the route(s) or just trying to hit the url?  And the resource line goes in your applications config/routes.yaml file.

**Reply by Niiph** [reply]
@ahundiak what's your `composer.json` autoload config?

**Reply by ahundiak** [reply]
Nothing special and I'd say it is irrelevant to routing.  Please don't tell me you are getting a big error message about controllers not being found but neglected to include it in your post.
```
    "autoload": {
        "psr-4": {
            "App\\": "src/",
            "Cerad\\MyBundle\\": "src-my-bundle/src"
        }
    },
```

**Reply by Niiph** [reply]
Nope. I am only unable to import file, nothing about controllers. 
```
Unable to find file "@xxxAsyncTaskBundle/config/routing.yaml" in @xxxAsyncTaskBundle/config/routing.yaml (which is being imported from "/srv/api/config/routes.yaml"). 
Make sure the "xxxAsyncTaskBundle" bundle is correctly registered and loaded in the application kernel class. 
If the bundle is registered, make sure the bundle path "@xxxAsyncTaskBundle/config/routing.yaml" is not empty.
```
When I move this file to different location (like `src/Resources/config`) it works just fine

**Reply by stof** [reply]
Can you share a reproducer for that ?

**Reply by ahundiak** [reply]
The error message strongly implies that routing.yaml does not exist.  Sort of thing you get with a typo or case sensitivity or something of that nature. [Here is a repo](https://github.com/cerad/s62bundle/blob/main/config/routes.yaml) with my test bundle.  It really does work.

**Reply by Niiph** [reply]
@stof 

Here's reproducer

https://github.com/Niiph/ruting/tree/main/skeleton

**Reply by ahundiak** [reply]
```
class AsyncTaskBundle extends AbstractBundle
```
That will get the routes to work but then you will have to adjust the DI stuff to use the AbstractBundle's methods.

You can take a look at AbstractBundle::getPath to get an idea of the difference in paths.  In fact if you really wanted to stick with the older Bundle class then I suppose copying the AbstractBundle::getPath code to your bundle's class would work as well.

**Reply by Niiph** [reply]
So there was a trick to this after all :D Thank you!
